pkg/ds/farm.go

```diff
@@ -0,0 +1,358 @@
+					continue
+				}
+				dsf.children[newDS.ID] = dsf.spawnDaemonSet(&newDS, quitCh)
+			} else {
+				dsf.children[dsFields.ID] = dsf.spawnDaemonSet(dsFields, quitCh)
+			}
+		}
+	}
+
+	if len(changes.Updated) > 0 {
+		dsf.logger.Infof("The following daemon sets have been updated:")
+		for _, dsFields := range changes.Updated {
+			dsf.logger.Infof("%v", *dsFields)
+
+			// If the daemon set contends with another daemon set, disable it
+			// if that fails, delete it, if that fails too this is a fatal error
+			if _, ok := dsf.children[dsFields.ID]; ok {
+				dsIDContended, isContended, err := dsf.dsContends(dsFields)
+				if err != nil {
+					dsf.logger.Errorf("Error occurred when trying to check for daemon set contention: %v", err)
+					continue
+				}
+
+				if isContended {
+					dsf.logger.Errorf("Updated daemon set '%s' contends with %s", dsFields.ID, dsIDContended)
+					newDS, err := dsf.dsStore.Disable(dsFields.ID)
+					if err != nil {
+						dsf.logger.Errorf("Error occurred when trying to delete daemon set: %v", err)
+						continue
+					}
+					dsf.children[newDS.ID].updatedCh <- &newDS
+				} else {
+					dsf.children[dsFields.ID].updatedCh <- dsFields
+				}
+			}
+		}
+	}
+
+	if len(changes.Deleted) > 0 {
+		dsf.logger.Infof("The following daemon sets have been deleted:")
+		for _, dsFields := range changes.Deleted {
+			dsf.logger.Infof("%v", *dsFields)
+			if child, ok := dsf.children[dsFields.ID]; ok {
+				child.deletedCh <- dsFields
+			}
+		}
+	}
+}
+
+// Naive implementation of a guard, this checks if any of the scheduled nodes
+// are used two daemon sets, does not pre-emptively catch overlaps by selector
+// because of how kubernetes selectors work
+//
+// Also naively checks the selectors to see if there are any selector overlap
+// if two label selectors are labels.Everything()
+//
+// Returns [ daemon set contended, contention exists, error ]
+func (dsf *Farm) dsContends(dsFields *ds_fields.DaemonSet) (ds_fields.ID, bool, error) {
+	// This daemon set does not contend if it is disabled
+	if dsFields.Disabled {
+		return "", false, nil
+	}
+	// Get all eligible nodes for this daemon set by looking at the labes.NODE tree
+	eligibleNodes, err := dsf.scheduler.EligibleNodes(dsFields.Manifest, dsFields.NodeSelector)
+	if err != nil {
+		return "", false, util.Errorf("Error retrieving eligible nodes for daemon set: %v", err)
+	}
+
+	// If this daemon set has a node selector set to Everything, check the labels
+	// of other daemon sets
+	for _, child := range dsf.children {
+		everythingSelector := klabels.Everything().String()
+		if !child.ds.IsDisabled() && child.ds.PodID() == dsFields.PodID && child.ds.ID() != dsFields.ID {
+			// Naively check if both selectors are the Everything selector plus
+			// something else or if both selectors are the same
+			//
+			// It is possible that both selectors won't contend if they both have
+			// something like { az = zone_one , az != zone_one } and that this
```

Comment:
it looks to me like this comment isn't true, this is just checking for klabels.Everything() which I believe is represented with an empty string